src/swap/do_action.rs

```diff
@@ -0,0 +1,212 @@
+use crate::{swap::db, SwapId};
+use comit::Timestamp;
+
+#[async_trait::async_trait]
+pub trait Do<E>
+where
+    Self: CheckMemory<E> + ShouldAbort + Execute<E> + Remember<E>,
+    E: Clone + Send + Sync + 'static,
+    <Self as Execute<E>>::Args: Send + Sync,
+{
+    async fn r#do(
```

Comment:
what is that? is it because `do` is a Rust language keyword?

Reply:
Yeah. I can move away from this name anyway, it's not final.

Reply:
Settled on `DoItOnce` as you suggested; coming with next PR.